text/0000-dockerfiles.md

```diff
@@ -0,0 +1,228 @@
+# Meta
+[meta]: #meta
+- Name: Support Dockerfiles
+- Start Date: 2021-06-30
+- Author(s): sclevine
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: [RFC0069](https://github.com/buildpacks/rfcs/blob/main/text/0069-stack-buildpacks.md), [RFC#167](https://github.com/buildpacks/rfcs/pull/167)
+- Depends on: [RFC#172](https://github.com/buildpacks/rfcs/pull/172)
+
+# Summary
+[summary]: #summary
+
+This RFC introduces functionality for customizing base images, as an alternative to stackpacks.
+
+# Motivation
+[motivation]: #motivation
+
+Relying on Dockerfiles for base image generation and manipulation allows us to apply buildpacks only to the problem that they solve best: managing application runtimes and dependencies.
+
+# What it is
+[what-it-is]: #what-it-is
+
+This RFC proposes that we replace stackpacks with build-time and runtime Dockerfiles that act as pre-build hooks.
+These hooks would be equivalent to base image / builder customizations, but would execute immediately before a build.
+
+Most importantly: the changes below do not impact buildpacks or the buildpack API.
+For a given application, a build that uses hooks could be optimized by creating a more narrowly-scoped builder with the hooks pre-applied, without modifying the buildpacks.
+
+# How it Works
+[how-it-works]: #how-it-works
+
+Note: kaniko, buildah, BuildKit, or the original Docker daemon may be used to apply Dockerfiles at the platform's discretion. 
+
+### Dynamically-applied Dockerfiles
+
+A builder image may include any number of "hook" files in `/cnb/hooks.d/`. Hooks may be associated with zero or more buildpack IDs.
```

Comment:
How a buildpack builder will be able to figure out he different hooks available, how they should be linked to a buildpack,...

**Remark**:  This is why I was mentioning within the `what it is` remark that a dependency exits between a hook and buildpack

Reply:
> How a buildpack builder will be able to figure out he different hooks available, how they should be linked to a buildpack,...

A buildpack can inspect the base image to determine if hooks have made changes (e.g., could run `python -v` to determine if python is available or needs a `require`)

> Remark: This is why I was mentioning within the what it is remark that a dependency exits between a hook and buildpack

A goal of this RFC is that no explicit dependency exists between hooks and buildpacks. Hooks are just a way of creating narrowly-scoped builder right before the buildpack build process.

Reply:
After thinking about this more, since the hooks would apply after the detect phase in the most recent proposal, the buildpack wouldn't be able to adjust the `require`. So I agree -- need to think about this more